disnake/http.py

```diff
@@ -188,69 +197,458 @@ def bucket(self) -> str:
+#         self.proxy: Optional[str] = proxy
+#         self.proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
+#         self.use_clock: bool = not unsync_clock
+#
+#         user_agent = "DiscordBot (https://github.com/DisnakeDev/disnake {0}) Python/{1[0]}.{1[1]} aiohttp/{2}"
+#         self.user_agent: str = user_agent.format(__version__, sys.version_info, aiohttp.__version__)
+
+
 class HTTPClient:
-    """Represents an HTTP client sending HTTP requests to the Discord API."""
+    """Represents an HTTP client for sending HTTP requests to and handling the rate limits of the Discord API.
+    Also, not thread safe.
+
+    Parameters
+    ----------
+    connector
+    default_max_per_second: :class:`int`
+        Maximum amount of requests per second per authorization.
+        Discord by default only allows 50 requests per second, but if your bot has had its maximum increased, then
+        increase this parameter.
+    time_offset: :class:`float`
+        Amount of seconds added to all ratelimit timers for lag compensation.
+        Due to latency and Discord servers not perfectly time synced, having no offset can cause 429's to occur even
+        with us following the reported X-RateLimit-Reset-After.
+        Increasing will protect from erroneous 429s but will slow bucket resets, lowering max theoretical speed.
+        Decreasing will hasten bucket resets and increase max theoretical speed but may cause 429s.
+    default_auth: Optional[:class:`str`]
+        Default string to use in the Authorization header if it's not manually provided.
+    proxy
+    proxy_auth
+    loop
+    """
 
     def __init__(
         self,
         connector: Optional[aiohttp.BaseConnector] = None,
         *,
-        loop: asyncio.AbstractEventLoop,
+        default_max_per_second: int = 50,
+        time_offset: float = 0.0,
+        default_auth: Optional[str] = None,
         proxy: Optional[str] = None,
         proxy_auth: Optional[aiohttp.BasicAuth] = None,
-        unsync_clock: bool = True,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        # dispatch: Callable,
     ) -> None:
-        self.loop: asyncio.AbstractEventLoop = loop
-        self.connector = connector
-        self.__session: aiohttp.ClientSession = MISSING  # filled in static_login
-        self._locks: weakref.WeakValueDictionary[str, asyncio.Lock] = weakref.WeakValueDictionary()
-        self._global_over: asyncio.Event = asyncio.Event()
-        self._global_over.set()
-        self.token: Optional[str] = None
-        self.bot_token: bool = False
-        self.proxy: Optional[str] = proxy
-        self.proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
-        self.use_clock: bool = not unsync_clock
+        # TODO: Think about adding ratelimit_multiplier? Would reduce the internal RateLimit.limit by that
+        #  float (such as 0.7) and could allow people to run multiple NC bots/processes on the same token while avoiding
```

Comment:
```suggestion
        #  float (such as 0.7) and could allow people to run multiple DS bots/processes on the same token while avoiding
```

Reply:
Perhaps just "Disnake"? I don't think "DS" is a recognised alias for disnake (or at least i would've hardly recognised it here without the direct comparison of nextcord/disnake).